File: leetcode-practice/java/code/54. 替换数字（第八期模拟笔试）.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // sc.next() 只会读取输入的第一个单词
        String s = sc.next();

        // 获取输入字符串的长度
        int strLen = s.length();

        // 遍历字符串中的每一个字符，统计数字字符的数量
        for (int i = 0; i < s.length(); i++) {

            // 检查当前字符是否是数字（ASCII范围 '0' 到 '9'）
            if (s.charAt(i) >= '0' && s.charAt(i) <= '9') {
                // 如果是数字，增加len的值，因为每个数字将被替换为"number"（6个字符）
                strLen += 5; // 原字符占1个位置，替换后增加5个位置
            }
        }

        // 创建一个新的字符数组，用于存储替换后的结果
        // 数组的长度为原字符串长度加上所有数字被替换后增加的长度
        char[] stringCharArray = new char[strLen];
        // 将原字符串s中的每个字符依次复制到新数组的前部
        for (int i = 0; i < s.length(); i++) {
            stringCharArray[i] = s.charAt(i);
        }

        // 使用双指针方法，从后往前遍历并替换数字字符
        // i指向原字符串的当前字符，j指向新数组中下一个可写入的位置
        int strOldIndex = s.length() - 1;
        int strNewIndex = strLen - 1;
        while (strOldIndex >= 0) {
            if ('0' <= stringCharArray[strOldIndex] && stringCharArray[strOldIndex] <= '9') {
                stringCharArray[strNewIndex--] = 'r';
                stringCharArray[strNewIndex--] = 'e';
                stringCharArray[strNewIndex--] = 'b';
                stringCharArray[strNewIndex--] = 'm';
                stringCharArray[strNewIndex--] = 'u';
                stringCharArray[strNewIndex--] = 'n';
            } else {
                stringCharArray[strNewIndex--] = stringCharArray[strOldIndex];
            }

            strOldIndex--;
        }

        System.out.println(new String(stringCharArray));
    }
}

File: leetcode-practice/cpp/code/54. 替换数字（第八期模拟笔试）.cpp
#include <iostream>

using namespace std;

int main() {
    string s; // 定义字符串变量s
    while (cin >> s) {
        // 循环读取输入流中的字符串（以空格分隔），直到输入结束
        // 记录原字符串末尾索引（从0开始计数）
        int strOldIndex = s.size() - 1;
        // 初始化数字计数器为0
        int count = 0;

        // 遍历字符串，统计数字字符的个数
        for (int i = 0; i < s.size(); i++) {
            // 判断字符是否为数字码范围判断）
            if (s[i] >= '0' && s[i] <= '9') {
                // 数字字符个数加1
                count++;
            }
        }

        int newLength = s.size() + count * 5;
        s.resize(newLength); // 使用resize调整大小
        int strNewIndex = newLength - 1; // 从真实末尾开始写入

        // 从末尾开始遍历原字符串
        // 逆序处理保证原字符串前半部分不受影响
        while (strOldIndex >= 0) {
            if (s[strOldIndex] >= '0' && s[strOldIndex] <= '9') {
                // 遇到数字字符，替换为"number"
                s[strNewIndex--] = 'r';
                s[strNewIndex--] = 'e';
                s[strNewIndex--] = 'b';
                s[strNewIndex--] = 'm';
                s[strNewIndex--] = 'u';
                s[strNewIndex--] = 'n';
            } else {
                // 遇到非数字字符，原字符直接复制到新字符串末尾
                s[strNewIndex--] = s[strOldIndex];
            }

            strOldIndex--;
        }

        cout << s << endl;
    }

    return 0;
}
